cli/src/providers/azure/InfrastructureBuilder.py

```diff
@@ -220,13 +221,16 @@ def get_storage_share_config(self):
         storage_share.specification.storage_account_name = storage_account_name(self.cluster_prefix, self.cluster_name, 'k8s')
         return storage_share
 
-    def get_vm(self, component_key, vm_config, availability_set, network_interface_name, security_group_association_name, index):
+    def get_vm(self, component_key, alt_component_name, vm_config, availability_set, network_interface_name, security_group_association_name, index):
         vm = dict_to_objdict(deepcopy(vm_config))
-        vm.specification.name = resource_name(self.cluster_prefix, self.cluster_name, 'vm' + '-' + str(index), component_key)
+        host_component_key = alt_component_name if alt_component_name and alt_component_name.strip() else component_key
+        vm.specification.name = resource_name(self.cluster_prefix, self.cluster_name, 'vm' + '-' + str(index), host_component_key)
         if self.hostname_domain_extension != '':
-            vm.specification.hostname = resource_name(self.cluster_prefix, self.cluster_name, 'vm' + '-' + str(index) + f'.{self.hostname_domain_extension}', component_key)
+            vm.specification.hostname = resource_name(self.cluster_prefix, self.cluster_name, 'vm' + '-' + str(index) + f'.{self.hostname_domain_extension}', host_component_key)
         else:
             vm.specification.hostname = vm.specification.name
+        if len(vm.specification.hostname) > 63:
```

Comment:
Let's not use magical consts, I suggest defining HOST_NAME_MAX_LENGTH const and using here and line below.